dimod/generators/mimo.py

```diff
@@ -114,89 +168,122 @@ def _symbols_to_spins(symbols: np.array, modulation: str) -> np.array:
         if symbols.ndim == 1:    # If symbols shaped as vector, return as vector
             spins.reshape((len(spins), ))
-        # elif symbols.shape[0] == 1:   #Jack: I think this is already baked in
-        #     spins.reshape((1, len(spins)))
-        # elif symbols.shape[1] == 1:
-        #     spins.reshape((len(spins), 1))
-        # else:   # Leave for manual reshaping
-        #     pass 
-    return spins
 
+    return spins
 
 def _yF_to_hJ(y, F, modulation):
-    offset, h, J = _quadratic_form(y, F) # Quadratic form re-expression
-    h, J = _real_quadratic_form(h, J, modulation) # Complex symbols to real symbols (if necessary)
-    h, J = _amplitude_modulated_quadratic_form(h, J, modulation) # Real symbol to linear spin encoding
+    """Convert :math:`O(v) = ||y - F v||^2` to modulated quadratic form.
+    
+    Constructs coefficients for the form 
+    :math:`O(v) = v^{\dagger} J v - 2 \Re(h^{\dagger} v) + \\text{offset}`. 
+
+    Args:
+        y: Received symbols as a NumPy column vector of complex or real values.
+
+        F: Wireless channel as an :math:`i \\times j` NumPy matrix of complex 
+            values, where :math:`i` rows correspond to :math:`y_i` receivers 
+            and :math:`j` columns correspond to :math:`v_i` transmitted symbols.
+
+        modulation: Modulation. Supported values are non-quadrature modulation 
+            BPSK and quadrature modulations 'QPSK', '16QAM', '64QAM', and '256QAM'.
+    
+    Returns:
+        Three tuple of amplitude-modulated linear biases :math:`h`, as a NumPy
+        vector, amplitude-modulated quadratic interactions, :math:`J`, as a  
+        matrix, and offset as a real scalar.
+    """
+    offset, h, J = _quadratic_form(y, F) # Conversion to quadratic form 
+
+    # Separate real and imaginary parts of quadratic form: 
+    h, J = _real_quadratic_form(h, J, modulation) 
+
+    # Amplitude-modulate the biases in the quadratic form:
+    h, J = _amplitude_modulated_quadratic_form(h, J, modulation) 
+
     return h, J, offset
 
 def linear_filter(F, method='zero_forcing', SNRoverNt=float('Inf'), PoverNt=1):
-    """ Construct linear filter W for estimation of transmitted signals.
-    # https://www.youtube.com/watch?v=U3qjVgX2poM
-   
+    """Construct a linear filter for estimating transmitted signals. 
     
-    We follow conventions laid out in MacKay et al. 'Achievable sum rate of MIMO MMSE receivers: A general analytic framework'
-    N0 Identity[N_r] = E[n n^dagger]
-    P/N_t Identify[N_t] = E[v v^dagger], i.e. P = constellation_mean_power*Nt for i.i.d elements (1,2,10,42)Nt for BPSK, QPSK, 16QAM, 64QAM.
-    N_r N_t = E_F[Tr[F Fdagger]], i.e. E[||F_{mu,i}||^2]=1 for i.i.d channel.  - normalization is assumed to be pushed into symbols.
-    SNRoverNt = PoverNt/N0 : Intensive quantity. 
-    SNRb = SNR/(Nt*bits_per_symbol)
-
-    Typical use case: set SNRoverNt = SNRb
-    """
+    # Jack: you'll need to go over the following carefully
```

Comment:
@jackraymond, to help you view this, here's an HTML rendition of the Latex:
![image](https://github.com/jackraymond/dimod/assets/34041130/861f8f4f-8dc7-49f8-b4e7-6a09d24b0a19)